warmrestart/warmRestartHelper.cpp

```diff
@@ -0,0 +1,358 @@
+#include <cassert>
+#include <sstream>
+
+#include "warmRestartHelper.h"
+
+
+using namespace swss;
+
+
+WarmStartHelper::WarmStartHelper(RedisPipeline      *pipeline,
+                                 ProducerStateTable *syncTable,
+                                 const std::string  &syncTableName,
+                                 const std::string  &dockerName,
+                                 const std::string  &appName) :
+    m_restorationTable(pipeline, syncTableName, false),
+    m_syncTable(syncTable),
+    m_syncTableName(syncTableName),
+    m_dockName(dockerName),
+    m_appName(appName)
+{
+    WarmStart::initialize(appName, dockerName);
+}
+
+
+WarmStartHelper::~WarmStartHelper()
+{
+}
+
+
+void WarmStartHelper::setState(WarmStart::WarmStartState state)
+{
+    WarmStart::setWarmStartState(m_appName, state);
+
+    /* Caching warm-restart FSM state in local member */
+    m_state = state;
+}
+
+
+WarmStart::WarmStartState WarmStartHelper::getState(void) const
+{
+    return m_state;
+}
+
+
+/*
+ * To be called by each application to obtain the active/inactive state of
+ * warm-restart functionality, and proceed to initialize the FSM accordingly.
+ */
+bool WarmStartHelper::isEnabled(void)
```

Comment:
I could not think of a good name for this function.  isEnabled() makes people feel it is safe to call it anytime they like to check the state, similar to isReconciled(void) or inProgress(void) , which is actually not true.   How about bool WarmStartHelper::checkAndStart(void)?

Reply:
Yes, i see your point. CheckAndStart() sounds good to me, will change.